dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java

```diff
@@ -2392,6 +2392,24 @@ public void setDataApprovalLevels( Map<OrganisationUnit, Integer> dataApprovalLe
         return ImmutableList.copyOf( ListUtils.union( getPeriods(), getFilterPeriods() ) );
     }
 
+    /**
+     * Returns all organisation units part of a dimension or filter.
+     */
+    public List<DimensionalItemObject> getAllOrganisationUnits()
+    {
+        return ImmutableList.copyOf( ListUtils.union( getOrganisationUnits(), getFilterOrganisationUnits() ) );
+    }
+
+    /**
+     * Returns all typed organisation part of a dimension or filter.
+     */
+    public List<OrganisationUnit> getAllTypedOrganisationUnits()
+    {
+        return getAllOrganisationUnits().stream()
```

Comment:
Should this method return an immutable list as well?

Reply:
There really is no need since the list is already a new independent list. In later JDKs there will be `Collectors.toImmutableList()` to make that new list modifiable.

Reply:
Agree with both of you.

It is a slight improvement to fail fast if a caller tries to modify the list though to avoid confusion, I will update it.